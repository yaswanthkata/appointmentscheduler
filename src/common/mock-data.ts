import { Doctor } from "../types/Doctor";

export const MockDoctor: Doctor = {
  profile: {
    name: "LÃ©a Pereira",
    rating: 4,
    location: "Lisbon",
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam a mauris
    laoreet, accumsan magna sit amet, dapibus augue. Donec enim dolor,
    luctus in faucibus eu, viverra sit amet dolor. Proin eu viverra mi, ac
    consectetur nunc. Ut aliquet diam nec sem sodales, vitae hendrerit purus
    finibus. Nulla posuere ac neque sed vestibulum. Nullam et euismod risus.
    Nam ac dignissim justo. Vestibulum volutpat maximus orci nec ultricies.`,
    thumbnail:
      "https://images.unsplash.com/photo-1513721032312-6a18a42c8763?w=152&h=152&fit=crop&crop=faces",
    rate: "$150",
    specialization: "PSICOLOGIST"
  },
  availableDays: [
    {
      calendarDay: "2019-08-25",
      slots: [
        { time: "2019-08-25T08:00:00+0530", isAvailable: false },
        { time: "2019-08-25T08:30:00+0530", isAvailable: false },
        { time: "2019-08-25T09:00:00+0530", isAvailable: false },
        { time: "2019-08-25T09:30:00+0530", isAvailable: false },
        { time: "2019-08-25T10:00:00+0530", isAvailable: false }
      ]
    },
    {
      calendarDay: "2019-08-26",
      slots: [
        { time: "2019-08-26T08:00:00+0530", isAvailable: true },
        { time: "2019-08-26T08:30:00+0530", isAvailable: true },
        { time: "2019-08-26T09:00:00+0530", isAvailable: true },
        { time: "2019-08-26T09:30:00+0530", isAvailable: true },
        { time: "2019-08-26T10:00:00+0530", isAvailable: true }
      ]
    },
    {
      calendarDay: "2019-08-27",
      slots: [
        { time: "2019-08-27T08:00:00+0530", isAvailable: true },
        { time: "2019-08-27T08:30:00+0530", isAvailable: true },
        { time: "2019-08-27T09:00:00+0530", isAvailable: true },
        { time: "2019-08-27T09:30:00+0530", isAvailable: true },
        { time: "2019-08-27T10:00:00+0530", isAvailable: true }
      ]
    },
    {
      calendarDay: "2019-08-28",
      slots: [
        { time: "2019-08-28T08:00:00+0530", isAvailable: true },
        { time: "2019-08-28T08:30:00+0530", isAvailable: true },
        { time: "2019-08-28T09:00:00+0530", isAvailable: true },
        { time: "2019-08-28T09:30:00+0530", isAvailable: true },
        { time: "2019-08-28T10:00:00+0530", isAvailable: true }
      ]
    },
    {
      calendarDay: "2019-08-29",
      slots: [
        { time: "2019-08-29T08:00:00+0530", isAvailable: true },
        { time: "2019-08-29T08:30:00+0530", isAvailable: true },
        { time: "2019-08-29T09:00:00+0530", isAvailable: true },
        { time: "2019-08-29T09:30:00+0530", isAvailable: true },
        { time: "2019-08-29T10:00:00+0530", isAvailable: true }
      ]
    }
  ]
};
