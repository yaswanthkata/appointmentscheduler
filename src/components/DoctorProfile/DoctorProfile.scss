/* Profile Section */
@import "../../styles/variables";
.profile {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  .profile-image {
    float: left;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    margin-right: 2rem;
    img {
      border-radius: 50%;
    }
  }
  .user-details div {
    margin-bottom: 5px;
  }
  .profile-user-name {
    font-size: 1.5rem;
    font-weight: 200;
  }
  .profile-specialization {
    font-weight: 400;
    color: $color-primary;
  }
  .rate{
    font-weight: bolder;
  }
}
.profile-bio {
  line-height: 1.5;
  padding: 10px;
  max-width: 90%;
  opacity: 0.8;
}
