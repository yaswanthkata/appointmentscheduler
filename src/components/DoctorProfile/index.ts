export { default as DoctorProfile } from "./DoctorProfile";
