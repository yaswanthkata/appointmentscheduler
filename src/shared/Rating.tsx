import * as React from "react";
import "./Rating.scss";

export interface IRatingProps {
  rating: number;
  total: number;
}

export default function Rating({ rating, total }: IRatingProps) {
  return (
    <>
      {[...Array(total)].map((n, i) => (
        <span
          key={i}
          className={i < rating ? "fa fa-star checked" : "fa fa-star"}
        />
      ))}
    </>
  );
}
