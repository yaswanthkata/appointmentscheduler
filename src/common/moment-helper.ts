import moment, { Moment } from "moment";
import { MomentSlot, CalendarDay } from "../types";
import { MockDoctor } from "./mock-data";

export const getDaySlots = (day: Moment): MomentSlot[] => {
  let convertedSlots: MomentSlot[] = [];
  const availableDay = MockDoctor.availableDays.find(
    d => day.format("YYYY-MM-DD") === d.calendarDay
  );
  if (availableDay) {
    convertedSlots = availableDay.slots.map(slot => {
      return {
        time: moment(slot.time),
        isAvailable:slot.isAvailable
      };
    });
  }
  return convertedSlots;
};

export const getFirstAvailableStartDate = (dates: CalendarDay[]) => {
  const moments: Moment[] = [];
  dates.forEach(d => moments.push(moment(d.calendarDay, "YYYY-MM-DD")));
  return moment.min(moments);
};
