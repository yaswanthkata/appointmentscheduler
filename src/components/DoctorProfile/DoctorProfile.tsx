import * as React from "react";
import { Rating } from "../../shared";
import { Profile } from "../../types";
import "./DoctorProfile.scss";

export interface IDoctorProfileProps {
  profile: Profile;
}

export default function DoctorProfile({ profile }: IDoctorProfileProps) {
  return (
    <>
      <div className="profile">
        <div className="profile-image">
          <img src={profile.thumbnail} alt="" />
        </div>
        <div className="user-details">
          <div className="profile-user-name">{profile.name}</div>
          <div>
            <span className="profile-specialization">
              {profile.specialization}
            </span>{" "}
            | <span>{profile.location}</span>
          </div>
          <div>
            <span>
              <Rating rating={profile.rating} total={5} />
              (20 Reviews)
            </span>
          </div>
          <div>
            <span className="rate">{profile.rate}</span> / 50 Minutes
          </div>
        </div>
      </div>
      <p className="profile-bio">{profile.description}</p>
    </>
  );
}
