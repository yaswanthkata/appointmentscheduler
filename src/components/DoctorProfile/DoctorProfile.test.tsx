import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import DoctorProfile, { IDoctorProfileProps } from './DoctorProfile';
import { Rating } from '../../shared';
import { MockDoctor } from '../../common/mock-data';

const props: IDoctorProfileProps = {
   profile:MockDoctor.profile
}

let componentWrapper: ShallowWrapper<React.Component>;

beforeEach(() => {
    componentWrapper = shallow(<DoctorProfile {...props} />);
});

it('should render profile component correctly', () => {
    expect(componentWrapper).toMatchSnapshot();
});

it('should render a profile thumbnail', () => {
    expect(componentWrapper.find('img')).toHaveLength(1);
});

it("should render  a rating component", () => {
    const ratingComponent = componentWrapper.find(Rating).length;

    expect(ratingComponent).toEqual(1);
});

it("should render profile-description", () => {
    expect(componentWrapper.find('.profile-bio')).toHaveLength(1);
});