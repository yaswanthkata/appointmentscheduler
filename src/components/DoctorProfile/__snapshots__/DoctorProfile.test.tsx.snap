// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render profile component correctly 1`] = `ShallowWrapper {}`;
